app/scripts/controllers/globalPromotions.ts

```diff
@@ -138,6 +144,53 @@ class GlobalPromotionsCtrl {
   isActive(promotion: GlobalPromotion): boolean {
     return this.globalPromotionService.isPromotionActive(promotion);
   }
+
+  setOrder(field: string) {
+    if (this.orderByField === field) {
+      this.reverseSort = !this.reverseSort;
+    } else {
+      this.orderByField = field;
+      this.reverseSort = false;
+    }
+  }
+
+  getStatusForSorting(promotion: GlobalPromotion): string {
+    return this.isActive(promotion) ? 'Active' : 'Inactive';
+  }
+
+  get sortField(): string | ((promotion: GlobalPromotion) => string) {
+    return this.orderByField === 'status'
+      ? this.getStatusForSorting
+      : this.orderByField;
+  }
+
+  get filteredPromotions(): GlobalPromotion[] {
```

Comment:
Using getters for this is a great idea! It makes the HTML a little cleaner than `getFilteredPromotions()`.